airflow/models.py

```diff
@@ -3032,6 +3032,25 @@ def get_task_instances(
     def roots(self):
         return [t for t in self.tasks if not t.downstream_list]
 
+    def topological_sort(self):
+        """
+        Sorts tasks in topographical order, such that a task comes after any of its
+        upstream dependencies.
+        """
+        stack, visited, sorted_tasks = [], set(), []
+
+        # begin with any tasks that have no upstream dependency
+        stack.extend(t for t in self.tasks if len(t.upstream_list) == 0)
+
+        while stack:
```

Comment:
There is an edge case that this wouldn't hit (and we should add it to tests as well).

If a DAG looks like:
A->B
A->C->D
(where A -> B means task A depends on task B)
then your algorithm will have the stack looking like this over time:
B
B D
D A (INCORRECT! A should not be processed until C is processed).

Also visited is not being used here.